catalog/solr/catalog-solr-offline-gazetteer/src/main/java/ddf/catalog/solr/offlinegazetteer/CatalogGazetteerForwardingPlugin.java

```diff
@@ -0,0 +1,233 @@
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PostIngestPlugin;
+import ddf.catalog.plugin.PreQueryPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.common.SolrInputDocument;
+import org.codice.solr.client.solrj.SolrClient;
+import org.codice.solr.factory.SolrClientFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CatalogGazetteerForwardingPlugin implements PostIngestPlugin, PreQueryPlugin {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(CatalogGazetteerForwardingPlugin.class);
+
+  private final SolrClient solrClient;
+
+  public CatalogGazetteerForwardingPlugin(SolrClientFactory clientFactory) {
+    synchronized (clientFactory) {
+      this.solrClient = clientFactory.newClient(COLLECTION_NAME);
+      Boolean response =
+          Failsafe.with(
+                  new RetryPolicy()
+                      .retryWhen(false)
+                      .withMaxDuration(5, TimeUnit.SECONDS)
+                      .withBackoff(25, 1_000, TimeUnit.MILLISECONDS))
+              .get(() -> this.solrClient.isAvailable());
+      if (response == null || !response) {
+        LOGGER.error("Unable to get Solr client for collection: {}", COLLECTION_NAME);
+      }
+    }
+  }
+
+  @Override
+  public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+    List<Metacard> gazetteerMetacards =
+        input
+            .getCreatedMetacards()
+            .stream()
+            .filter(this::isGazetteerMetacard)
+            .collect(Collectors.toList());
+
+    if (gazetteerMetacards.isEmpty()) {
+      return input;
+    }
+
+    try {
+      solrClient.add(
+          gazetteerMetacards
+              .stream()
+              .map(CatalogGazetteerForwardingPlugin::convert)
+              .collect(Collectors.toList()));
+    } catch (SolrServerException | IOException e) {
+      throw new PluginExecutionException("Error while processing gazetteer data", e);
+    }
+
+    return input;
+  }
+
+  @Override
+  public UpdateResponse process(UpdateResponse input) throws PluginExecutionException {
+    List<Metacard> gazetteerMetacards =
+        input
+            .getUpdatedMetacards()
+            .stream()
+            .map(Update::getNewMetacard)
+            .filter(this::isGazetteerMetacard)
+            .collect(Collectors.toList());
+
+    if (gazetteerMetacards.isEmpty()) {
+      return input;
```

Comment:
would it be useful to log something here?

Reply:
I think something at trace could be helpful.